### PR TITLE
Update to symfony7

### LINKED ISSUE #5950: Add support for symfony 7 in 13.x and minimum version of PHP 8.3
**Is your feature request related to a problem? Please describe.**
Drupal 11.x now requires version 7 of symfony/* components so drush/drush cannot be installed alongside it
Drupal 11.x now requires PHP 8.3

**Describe the solution you'd like**
Pre-reqs
* https://github.com/consolidation/robo/pull/1156
* https://github.com/consolidation/annotated-command/pull/307
* https://github.com/consolidation/config/pull/62
* https://github.com/consolidation/log/pull/35
* https://github.com/consolidation/site-process/pull/77
* https://github.com/consolidation/site-alias/pull/70
* https://github.com/consolidation/self-update/pull/32
* https://github.com/grasmash/yaml-cli/pull/23

Support 7.x versions of symfony/*

### PR DESCRIPTION
**Pull request by larowlan**
Fixes #5950

## Prereqs
* https://github.com/consolidation/robo/pull/1156
* https://github.com/consolidation/annotated-command/pull/307
* https://github.com/consolidation/config/pull/62
* https://github.com/consolidation/log/pull/35
* https://github.com/consolidation/site-process/pull/77
* https://github.com/consolidation/site-alias/pull/70
* https://github.com/consolidation/self-update/pull/32
* https://github.com/grasmash/yaml-cli/pull/23

### CODE REVIEW

File: src-symfony-compatibility/v6/Symfony/LessStrictArgvInput.php [hunk 1]
```diff
@@ -58,15 +58,15 @@ public function getOption(string $name): mixed
         return false;
     }
 
-    protected function setTokens(array $tokens)
+    protected function setTokens(array $tokens): void
```
Comment (greg-1-anderson) [inline src-symfony-compatibility/v6/Symfony/LessStrictArgvInput.php:61]: The path to this file still says `v6`. Maybe this should be in a `v7` folder for clarity.

File: src/Preflight/Preflight.php [hunk 2]
```diff
@@ -206,6 +206,8 @@ public function loadSymfonyCompatabilityAutoloader(): ClassLoader
             4 => 'v4',  // Drupal 9
             5 => 'v4',  // Early Drupal 10 (Symfony 5 works with Symfony 4 classes, so we don't keep an extra copy)
             6 => 'v6',  // Drupal 10
+            // @TODO Revisit if needed.
+            7 => 'v6',  // Drupal 11
```
Comment (greg-1-anderson) [inline src/Preflight/Preflight.php:210]: Should point to v7, interface is different for v6 vs v7. Could probably remove this mechanism if Drush only supported one major version of Drupal at a time, but it seems useful to support multiple versions.

**Comment by weitzman** [comment]
Almost there. Robo tag is imminent. This will be green once we get https://github.com/wodby/php/issues/194

**Comment by greg-1-anderson** [comment]
@weitzman: Can we merge with disabled (or failing) sqlite tests, and follow up after https://github.com/wodby/php/issues/194 (or Alpine 3.20) is available?

File: src/Preflight/Preflight.php (hunk 2 above)
Comment (greg-1-anderson) [reply src/Preflight/Preflight.php:210]: I can do this as follow-up work if you want to merge like this to get Symfony 7 into 13.x-dev

**Pull request merged**